cmd/akash/deployment.go

```diff
@@ -87,77 +105,186 @@ func createDeployment(session session.Session, cmd *cobra.Command, args []string
 		return err
 	}
 
-	res, err := txclient.BroadcastTxCommit(&types.TxCreateDeployment{
-		Tenant:   txclient.Key().GetPubKey().Address().Bytes(),
-		Nonce:    nonce,
-		OrderTTL: ttl,
-		Groups:   groups,
-		Version:  hash,
-	})
-
-	if err != nil {
-		session.Log().Error("error sending tx", "error", err)
-		return err
+	resChan := make(chan *tmctypes.ResultBroadcastTxCommit, 1)
+	errChan := make(chan error)
+	errC := make(chan error)
+	var res *tmctypes.ResultBroadcastTxCommit
+	var address []byte
+	var start time.Time
+	var elapsed time.Duration
+	blue := color.New(color.FgCyan, color.Bold)
+
+	go func() {
+		res, err := txclient.BroadcastTxCommit(&types.TxCreateDeployment{
+			Tenant:   txclient.Key().GetPubKey().Address().Bytes(),
+			Nonce:    nonce,
+			OrderTTL: ttl,
+			Groups:   groups,
+			Version:  hash,
+		})
+		if err != nil {
+			ses.Log().Error("error sending tx", "error", err)
+			errChan <- err
+			return
+		}
+		resChan <- res
+	}()
+
+	lines := []rune{'—', '\\', '|', '/'}
+	start = time.Now()
+outloop:
+	for {
+		select {
+		case res = <-resChan:
+			address = res.DeliverTx.Data
+			fmt.Fprintln(writer.Bypass(), "(success) deployment posted with id:", X(address))
+			break outloop
+		case err := <-errChan:
+			ses.Log().Error("error sending tx", "error", err)
+			return err
+		default:
```

Comment:
It looks like `default:` will never be triggered.  Maybe `time.After()`?  Would probably be easier to just do a separate gofunc that does the spinning, then stop it after reading from `resChan` or `errChan`.

Reply:
oh i see.  it'll be hit _constantly_.  yeah just put this in a separate thread that loops over a `time.Ticker`.